src/ray/common/source_location.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// A struct which represents source code location (aka. filename and line
+// number).
+
+#pragma once
+
+#include <iostream>
+#include <string_view>
+
+namespace ray {
+
+struct SourceLocation {
+  std::string_view filename;
```

Comment:
won't we have dangling pointer issue?

Reply:
No, string view is not POD.

Reply:
The only place to access should be status class and ostream printer

Reply:
Discussed offline, it's only safe because filename only points to `__FILE__` which won't be freed. We should mention that in the comment.

Reply:
Sure, comment added.